torch_xla/csrc/tensor_methods.cpp

```diff
@@ -2640,18 +2640,36 @@ XLATensorPtr trace(const XLATensorPtr& input) {
 
 XLATensorPtr transpose(const XLATensorPtr& input, int64_t dim0, int64_t dim1) {
   torch_xla::runtime::util::MaybeRef<xla::Shape> input_shape = input->shape();
-  ViewInfo view_info;
+
+  // See Note: [Disabling functionalization]
+  if (runtime::sys_util::GetEnvBool("XLA_DISABLE_FUNCTIONALIZATION", false)) {
+    ViewInfo view_info;
+    if (input_shape.get().rank() <= 1) {
+      // return a view of self if input rank <=1
+      torch::lazy::Value ir_value = input->GetIrValue();
+      view_info = ViewInfo(ViewInfo::Type::kNoOp, GetXlaShape(ir_value),
+                           GetXlaShape(ir_value));
+    } else {
+      std::vector<int64_t> permute_dims = torch::lazy::MakeTransposePermutation(
+          /*dim0=*/dim0, /*dim1=*/dim1, /*rank=*/input_shape.get().rank());
+      view_info = ViewInfo(ViewInfo::Type::kPermute, input_shape, permute_dims);
+    }
+    return input->CreateViewTensor(std::move(view_info));
+  }
+
+  XLATensorPtr result;
   if (input_shape.get().rank() <= 1) {
-    // return a view of self if input rank <=1
-    torch::lazy::Value ir_value = input->GetIrValue();
-    view_info = ViewInfo(ViewInfo::Type::kNoOp, GetXlaShape(ir_value),
-                         GetXlaShape(ir_value));
+    // kNoOp
+    result = input;
   } else {
+    // kPermute
     std::vector<int64_t> permute_dims = torch::lazy::MakeTransposePermutation(
         /*dim0=*/dim0, /*dim1=*/dim1, /*rank=*/input_shape.get().rank());
-    view_info = ViewInfo(ViewInfo::Type::kPermute, input_shape, permute_dims);
+    result = input->CreateFrom(torch::lazy::MakeNode<Permute>(
+        input->GetIrValue(), xla::InversePermutation(permute_dims)));
```

Comment:
Wondering why we need InversePermutation?

Reply:
maybe here? https://github.com/pytorch/xla/blob/1a4e43280181231c60e82e522caf3c0882531e1e/torch_xla/csrc/view.cpp#L93-L95

Reply:
Oh good catch, this should be actually without `xla:: InversePermutation`. I fixed it in the other PR for permute (https://github.com/pytorch/xla/pull/5195/commits/e37a9863d15cb99f764c797cd18c8f44ff9d4ba0), but must have missed it here. Let me update it. Thanks!